tests/test_query.py

```diff
@@ -0,0 +1,204 @@
+from pytest import raises
+
+dsl_query = 'op.fromView("test","musician")'
+serialized_query = '{"$optic":{"ns":"op", "fn":"operators", "args":[{"ns":"op", "fn":"from-view", "args":["test", "musician"]}]}}'
+sql_query = 'select * from musician'
+sparql_query = 'PREFIX musician: <http://marklogic.com/column/test/musician/> SELECT * WHERE {?s musician:lastName ?o}'
+
+
+def test_dsl_default(client):
+    data = client.rows.query(dsl_query)
+    assert type(data) is dict
+    assert 1 == len([c for c in data["columns"] if c["name"] == "test.musician.lastName"])
+    assert 1 == len([r for r in data["rows"] if r["test.musician.lastName"] == {'type': 'xs:string', 'value': 'Coltrane'}])
+
+
+def test_dsl_default_return_response(client):
+    response = client.rows.query(dsl_query, return_response=True)
+    assert 200 == response.status_code
+    data = response.json()
+    assert type(data) is dict
+    assert 1 == len([c for c in data["columns"] if c["name"] == "test.musician.lastName"])
+    assert 1 == len([r for r in data["rows"] if r["test.musician.lastName"] == {'type': 'xs:string', 'value': 'Coltrane'}])
+
+
+def test_query_bad_user(not_rest_user_client):
+    response = not_rest_user_client.rows.query(dsl_query)
+    assert 403 == response.status_code
+
+
+def test_dsl_json(client):
+    data = client.rows.query(dsl_query, format="json")
+    assert type(data) is dict
+    assert 1 == len([c for c in data["columns"] if c["name"] == "test.musician.lastName"])
+    assert 1 == len([r for r in data["rows"] if r["test.musician.lastName"] == {'type': 'xs:string', 'value': 'Coltrane'}])
+
+
+def test_dsl_xml(client):
+    data = client.rows.query(dsl_query, format="xml")
+    assert type(data) is str
+    assert '<t:columns>\n<t:column name="test.musician.lastName"/>' in data
+    assert '<t:cell name="test.musician.lastName" type="xs:string">Coltrane' in data
+
+
+def test_dsl_csv(client):
+    data = client.rows.query(dsl_query, format="csv")
+    assert type(data) is str
+    assert 'test.musician.lastName,' in data
+    assert '\r\nColtrane,John,1926-09-23\r\n' in data
+
+
+def test_dsl_json_seq(client):
+    data = client.rows.query(dsl_query, format="json-seq")
+    assert type(data) is str
+    rows = data.split("\n")
+    assert '{"columns":[{"name":"test.musician.lastName"},' in rows[0]
+    assert '"test.musician.lastName":{"type":"xs:string","value":"Coltrane"' in rows[1]
+
+
+def test_dsl_mixed(client):
+    response = client.rows.query(dsl_query, format="mixed")
+    data = response.json()
+    assert type(data) is dict
+    assert 1 == len([c for c in data["columns"] if c["name"] == "test.musician.lastName"])
+    assert 1 == len([r for r in data["rows"] if r["test.musician.lastName"] == {'type': 'xs:string', 'value': 'Coltrane'}])
+
+
+def test_serialized_default(client):
```

Comment:
So initially, I was thinking that since you cover every response type for DSL queries, we only need one test for each other input type. OTOH - maybe the server team would appreciate us having tests that verify that e.g. we get the same XML or CSV response regardless of whether the user submits a DSL or a serialized query? I can't think of any issues along those lines, so I lean towards dropping most of the tests below this in favor of just having one test per input type (while keeping all the tests above this for the DSL that verify all the output types).